### PR TITLE
Remove unnecessary pointers to vectors

### PR DESCRIPTION
**Pull request by hakonsbm**
This PR simplifies some vectors by using objects instead of pointers to objects wherever possible.

### CODE REVIEW

File: nestkernel/connection_manager.h [hunk 1]
```diff
@@ -561,7 +560,7 @@ class ConnectionManager : public ManagerInterface
    * connection information. Corresponds to a three dimensional
    * structure: threads|synapses|connections
    */
-  std::vector< std::vector< ConnectorBase* >* > connections_;
+  std::vector< std::vector< ConnectorBase* > > connections_;
```
Comment (jakobj) [inline nestkernel/connection_manager.h]: why is the innermost part still a pointer?

Comment (hakonsbm) [reply nestkernel/connection_manager.h]: I assumed that the synapses should have the possibility to be connectionless (i.e. have content `NULL`), or have a connection assigned to a specific `synapse_id`. But if that is not the case, I can remove this pointer too.

Comment (hakonsbm) [reply nestkernel/connection_manager.h]: Turns out the reason for `ConnectorBase` being a pointer here is that it is a base class for the template class `Connector`. By having this polymorphic class structure, the `connections_` vector can contain `Connector`s with different connection types. Therefore there is really no way to remove the pointer, at least not without making the code a lot more complicated and less general.

Comment (jakobj) [reply nestkernel/connection_manager.h]: OK great thanks for investigating! Makes sense, so I am happy with the current state.

**Pull request merged**